src/openMVG/voctree/database.cpp

```diff
@@ -62,20 +62,17 @@ void Database::sanityCheck(size_t N, std::vector< std::vector<Match> >& matches)
   matches.clear();
   // since we already know the size of the vectors, in order to parallelize the 
   // query allocate the whole memory
-  matches.resize(this->size(), std::vector<Match>(N));
   boost::progress_display display(database_.size());
   
-//  @todo JEME
-//  #pragma omp parallel for
-//  for(DocId i = 0; i < (DocId) database_.size(); ++i)
-//  for(auto& document: database_)
-//  {
-//    std::vector<Match> m;
-//    find(document.second, N, m);
-//    //		matches.emplace_back( m );
-//    matches[i] = m;
-//    ++display;
-//  }
+  //#pragma omp parallel for default(none) shared(database_)
```

Comment:
this may be achieved with an hack:
```c++
#pragma omp parallel shared(database_)
for(const auto & itData : database_)
{
  #pragma omp single nowait
  {
    std::vector<DocMatch> m;
    find(itData.second, N, m);
    #pragma omp critical
    {
      matches[itData.first] = m;
      ++display;
    }
  }
}
```
Not sure if it works (I never tried with the for range loop, with the iterator it works). Even if it works it is not optimal as there are a lot of overheads.
If you want you can test (both ways, with the range for and the iterator) otherwise we can just leave it as it is, as it is not that critical for the moment.